go-controller/hybrid-overlay/pkg/controller/master.go

```diff
@@ -0,0 +1,210 @@
+package controller
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	houtil "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/util"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/config"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/kube"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/ovn/allocator"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+
+	kapi "k8s.io/api/core/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+
+	. "github.com/onsi/ginkgo"
+)
+
+// MasterController is the master hybrid overlay controller
+type MasterController struct {
+	kube      *kube.Kube
+	allocator *allocator.SubnetAllocator
+}
+
+// NewMaster a new master controller that listens for node events
+func NewMaster(clientset kubernetes.Interface) (*MasterController, error) {
+	m := &MasterController{
+		kube:      &kube.Kube{KClient: clientset},
+		allocator: allocator.NewSubnetAllocator(),
+	}
+
+	// Add our hybrid overlay CIDRs to the allocator
+	for _, clusterEntry := range config.HybridOverlay.ClusterSubnets {
+		err := m.allocator.AddNetworkRange(clusterEntry.CIDR.String(), 32-clusterEntry.HostSubnetLength)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Mark existing hostsubnets as already allocated
+	existingNodes, err := m.kube.GetNodes()
+	if err != nil {
+		return nil, fmt.Errorf("Error in initializing/fetching subnets: %v", err)
+	}
+	for _, node := range existingNodes.Items {
+		hostsubnet, _ := houtil.ParseHybridOverlayHostSubnet(&node)
```

Comment:
If this returns an error, shouldn't you at least log it that the annotation is wrong?